tests/enrol_advanced_testcase_test.php

```diff
@@ -0,0 +1,256 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+/**
+ * Unit-Test for enrolment plugin
+ *
+ * @package    enrol_evento
+ * @copyright  2018 HTW Chur Thomas Wieling
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+defined('MOODLE_INTERNAL') || die();
+global $CFG;
+require_once($CFG->dirroot . '/enrol/evento/classes/task/evento_member_sync_task.php');
+//require_once($CFG->dirroot . '/enrol/evento/tests/local_evento_service_simulation.php');
+require_once($CFG->dirroot . '/local/evento/classes/evento_service.php');
+require_once($CFG->dirroot . '/enrol/evento/locallib.php');
+require_once($CFG->dirroot . '/enrol/evento/tests/locallib_exposed.php');
+
+
+ class mod_evento_advanced_testcase extends advanced_testcase {
+   /** @var stdClass Instance. */
+   private $instance;
+   /** @var stdClass Student. */
+   private $student;
+   /** @var stdClass First course. */
+   private $course1;
+   /** @var stdClass Second course. */
+   private $course2;
+   /** @var stdClass Second course. */
+   private $cat1;
+   /** @var stdClass Second course. */
+   private $cat2;
+   /** @var stdClass Plugin. */
+   private $plugin;
+   /** @var stdClass Plugin. */
+   private $locallib;
+   /** @var stdClass Plugin. */
+   private $user_enrolment;
+   /** @var stdClass Plugin. */
+   private $enrolments;
+
+   protected function setUp()
+   {
+     /*Create Moodle categories*/
+     $this->cat1 = $this->getDataGenerator()->create_category();
+     $this->cat2 = $this->getDataGenerator()->create_category();
+     /*Create Object $locallib*/
+     $this->locallib = new enrol_evento_user_sync_exposed();
+   }
+   /*Create courses*/
+   protected function create_moodle_course()
+   {
+     $this->course1 = $this->getDataGenerator()->create_course(array('category'=>$this->cat1->id, 'idnumber' => 'mod.mmpAUKATE1.HS18_BS.001'));
+   }
+   /*Enable plugin method*/
+   protected function enable_plugin()
+   {
+     $enabled = enrol_get_plugins(true);
+     $enabled['evento'] = true;
+     $enabled = array_keys($enabled);
+     set_config('enrol_plugins_enabled', implode(',', $enabled));
+   }
+   /*disable plugin method*/
+   protected function disable_plugin()
+   {
+     $enabled = enrol_get_plugins(true);
+     unset($enabled['evento']);
+     $enabled = array_keys($enabled);
+     set_config('enrol_plugins_enabled', implode(',', $enabled));
+   }
+   /*get enroled user from course*/
+   protected function get_enroled_user($id)
+   {
+     global $DB;
+     $this->user_enrolment = $DB->get_record('enrol', array('courseid'=>$id, 'enrol'=>'evento'), '*', MUST_EXIST);
+     $this->enrolments = $DB->count_records('user_enrolments', array('enrolid'=>$this->user_enrolment->id));
+   }
+   /*Basic test if plugin is enabled*/
+   public function test_basics()
+   {
+     $this->resetAfterTest(true);
```

Comment:
Der Codeteil
`$this->resetAfterTest(true);`
ist immer die erste Zeile. Das müsste sich mit setUp oder setUpBeforeClass machen lassen.
https://phpunit.de/manual/6.5/en/fixtures.html#fixtures.more-setup-than-teardown